### PR TITLE
Albertinad/editor update

### PR DESCRIPTION
**Pull request by albertinad**
Proposed changes to pumascript editor:
- removing uploaded libraries and start using bower to track front-end dependencies
- full support and use of requirejs in editor
- updating codemirror to latest available version on bower registry, that supports requirejs
- pumascript script is also loaded in the editor as a dependency, so it may be not necessary to require it on every execution/load, it is already in the editor context.

### CODE REVIEW

File: editor/editor.js [hunk 1]
```diff
@@ -1,69 +1,102 @@
...
+        var programStr = this.editorValue(this._pumaScriptEditor);
+
+        //Save the last execution in local storage
+        if (programStr !== '') {
+            localStorage.setItem("puma-backup", localStorage.getItem("puma"));
+            localStorage.setItem("puma", programStr);
+        }
+
+        if (programStr !== undefined && programStr !== null) {
+            result = puma.evalPuma(programStr);
```
Comment (albertinad) [inline editor/editor.js]: @emravera The only change to the implementation was at this function, the current implementation requires pumascript on each call, with this proposal, pumascript is loaded once and available always in the context. If you think that this is not right I will change it.

**Comment by emravera** [comment]
Great Work! I think we need to update README.md with the steps to initialize the repos dependencies or create a grunt "start-puma" fro this task. Maybe you want to include this issue in the tracker.

**Comment by albertinad** [comment]
Great! Yes, we should have some grunt tasks to initialize the repo easily, I updated the README.md for the editor and I will create an issue to track the grunt update.
Thanks!

**Comment by emravera** [comment]
Everything looks Ok I will merge the pull request. Thanks!

**Pull request merged**